lib/receiver.ts

```diff
@@ -0,0 +1,289 @@
+   * the "session_error" event that occurs on the underlying rhea session.
+   */
+  onSessionError?: rhea.OnAmqpEvent;
+  /**
+   * @property {rhea.OnAmqpEvent} [onSessionClose] The handler that can be provided for receiving the
+   * "session_close" event that occurs on the underlying rhea session.
+   */
+  onSessionClose?: rhea.OnAmqpEvent;
+}
+
+/**
+ * Describes the receiver that wraps the rhea receiver.
+ * @class Receiver.
+ */
+export class Receiver {
+  receiverOptions?: ReceiverOptions;
+  private _session: Session;
+  private _receiver: rhea.Receiver;
+
+  constructor(session: Session, receiver: rhea.Receiver, options?: ReceiverOptions) {
+    this._session = session;
+    this._receiver = receiver;
+    this.receiverOptions = options;
+  }
+
+  get name(): string {
+    return this._receiver.name;
+  }
+
+  get error(): rhea.AmqpError | Error | undefined {
+    return this._receiver.error;
+  }
+
+  get properties(): rhea.Dictionary<any> {
+    return this._receiver.properties;
+  }
+
+  get source(): rhea.Source {
+    return this._receiver.source;
+  }
+
+  get target(): rhea.TerminusOptions {
+    return this._receiver.target;
+  }
+
+  get address(): string {
+    return this.source.address;
+  }
+
+  get session(): Session {
+    return this._session;
+  }
+
+  get connection(): Connection {
+    return this._session.connection;
+  }
+
+  get drain(): boolean {
+    return this._receiver.drain;
+  }
+
+  addCredit(credit: number): void {
+    this._receiver.add_credit(credit);
+  }
+
+  setCreditWindow(creditWindow: number): void {
+    this._receiver.set_credit_window(creditWindow);
+  }
+  /**
+   * Determines whether the receiver link and its session is open.
+   * @returns {boolean} `true` open. `false` closed.
+   */
+  isOpen(): boolean {
+    let result = false;
+    if (this._session.isOpen() && this._receiver.is_open()) {
+      result = true;
+    }
+    return result;
+  }
+
+  /**
+   * Determines whether both local and remote endpoint for link or it's underlying session
+   * or it's underlying connection are closed.
+   * Within the "receiver_close", "session_close" event handler, if this
+   * method returns `false` it means that the local end is still open. It can be useful to
+   * determine whether the close was initiated locally under such circumstances.
+   *
+   * @returns {boolean} `true` if closed, `false` otherwise.
+   */
+  isClosed(): boolean {
+    return this._receiver.is_closed();
+  }
+
+  /**
+   * Determines whether both local and remote endpoint for just the link itself are closed.
+   * Within the "receiver_close" event handler, if this method returns `false` it
+   * means that the local end is still open. It can be useful to determine whether the close
+   * was initiated locally under such circumstances.
+   *
+   * @returns {boolean} `true` - closed, `false` otherwise.
+   */
+  isItselfClosed(): boolean {
```

Comment:
the wording of this is a little weird to me. would `isFullyClosed` make more sense? that's my understanding of what this means